app/models/taxon_change.rb

```diff
@@ -126,7 +128,15 @@ def active_children_conflict?
     end
     return true
   end
-    
+  
+  def is_branching?
+    taxon_change_taxa.map( &:taxon_id ).include? taxon_id
+  end
+
+  def need_not_push_to?( output_taxon_id )
```

Comment:
I don't know about you but I find negative boolean checks like this kind of confusing. How about

```ruby
def automatable_for_output?( output_taxon )
  output_taxon_id = output_taxon.is_a?( Taxon ) ? output_taxon.id : output_taxon
  if is_a?( TaxonSplit ) && is_branching?
    input_taxon.id == output_taxon_id
  else
    automatable?
  end
end
```

Then you'd need to negate this in most of your usage, e.g. `!taxon_change.automatable_for_output?( output_taxon )`, which also presents a readability challenge, but I think is slightly more readable. Personal preference I guess. If you prefer your current language, could you at least add a comment to the method to make it clear what you mean? Because "push" is not a verb we're currently using to describe the process of updating records associated with the taxa involved in a taxon change.

Reply:
I'm fine with that language but I want the method to do this
```ruby
def automatable_for_output?( output_taxon )
    output_taxon_id = output_taxon.is_a?( Taxon ) ? output_taxon.id : output_taxon
    if is_a?( TaxonSplit ) && is_branching?
      return input_taxon.id == output_taxon_id
    end
    return false
  end
```
because if its not a branching taxon split with the output under consideration matching the input , then I don't want to skip handling the input taxon content regardless of whether its automatable. So I'm not sure if adapting this term automatable for this new context adds confusion or not. But if you're ok with the above I'm fine with it

Reply:
If the change is a `TaxonSwap` or a `TaxonMerge`, then isn't it always automatable for the output? Or if it only ever applies to a `TaxonSplit`, maybe move it to that class and get rid of the type check in the method?

Reply:
I'm not trying to get at automatable here though - unless I'm misunderstanding
automatable in the past means is there a single output or are atlases in place to shuttle content to a single output
what I want here is 'is the output I'm considering the same taxon as the input for this taxon change?'
I'm fine with calling that 'automatable for output' I was just mentioning that it might be confusing since its kind of a different thing than how we use the term automatable elsewhere. Maybe 'in_place_for_output?' or something?

Reply:
> what I want here is 'is this output I'm considering the same taxon as the input for this taxon split?'

I understand, but what I'm getting at here is that it's not immediately clear *why* that is important. Like, if I'm reading this, `input_matches?( output taxon )` provides no more insight into the purpose of the check than `input_taxon == output_taxon` does, but naming the method `automatable_for_output?` implies the purpose of that check, i.e. "for this split, is it ok for us to automatically update records associated with the input for this output?" Generally the answer should be yes, but not if that output taxon is a part of a branching split and is the same as the input taxon.

Again, I'm picking nits and I don't mean for this to block your merge, so go ahead if you want. I'm just trying to preserve my sanity the next time I have to dig into this stuff.